### PR TITLE
fixed line-endings and added info on how to compile on cygwin

### PR DESCRIPTION
**Pull request by creighto**
Hi there! I tried to compile this on cygwin, but had to fix line-endings to get it to work. Unfortunately, github seemst to just ignore committed files that have ONLY changed line-endings. I added the exact commands that worked for me (some dependencies were missing on my machine) to the README.md

### CODE REVIEW

**Comment by versat** [comment]
Hi,
which file had the problematic line endings?
What are your git settings regarding autocrlf. Maybe that is the issue?
There is a CI Pipeline for Windows Cygwin builds, see appveyor.yml. It regularly verifies Cygwin builds, so it should work.

File: README.md [hunk 1]
```diff
@@ -97,17 +97,22 @@ broken RPM build environment. You should add this to your ~/.rpmmacros:
 
 ## Creating WINDOWS INSTALLER
 
-Install Cygwin and include at least the ghostscript, zip, dos2unix and libgcc
+Install Cygwin and include at least the gcc, make, cygrunsrv, ghostscript, zip, dos2unix and libgcc
```
Comment (fralken) [inline README.md:100]: Thanks for this. More precisely, it should be `gcc-core` rather than `gcc`, as exemplified later.

File: README.md [hunk 2]
```diff
@@ -97,17 +97,22 @@ broken RPM build environment. You should add this to your ~/.rpmmacros:
...
 packages.
 
 In case you are using a 64-bit version of Cygwin: rename cyggcc_s-1.dll to
 cyggcc_s-seh-1.dll in Makefile and win/setup.iss.
```
Comment (fralken) [inline README.md:105]: This line should be removed as well

File: README.md [hunk 3]
```diff
@@ -97,17 +97,22 @@ broken RPM build environment. You should add this to your ~/.rpmmacros:
...
 
+Install Cygwin using
+
+    setup-x86_64.exe -qgnNdO -l C:\cygwin64\var\cache\setup -R C:\cygwin64 -s http://cygwin.mirror.constant.com -P gcc-core -P make -P ghostscript -P dos2unix -P zip -P cygrunsrv
+
```
Comment (fralken) [inline README.md:109]: This is a bit opinionated. Options "n" and "N" disable creation of shortcuts, which contradict the instruction below. Probably the only relevant options are "-P" for installing packages.

File: README.md [hunk 4]
```diff
@@ -117,6 +122,11 @@ project definition file, installer:
     make win
 
 Now this automatically creates the installer.
+
+Alternative, run this command, which does these steps, too:
+
+    C:\cygwin64\bin\bash -e -l -c "cd /cygdrive/yourdrive/your_ctnlm_src_location && make distclean && ./configure && make DEBUG=1 && make win"
+
```
Comment (fralken) [inline README.md:129]: Option `DEBUG=1` is not relevant here

**Comment by creighto** [comment]
> Hi, which file had the problematic line endings? What are your git settings regarding autocrlf. Maybe that is the issue? There is a CI Pipeline for Windows Cygwin builds, see appveyor.yml. It regularly verifies Cygwin builds, so it should work.

The problematic files for me were the configure script and the doc/cntlm.1 files:
```
$ ./configure
./configure: line 13: $'\r': command not found
./configure: line 29: syntax error near unexpected token `$'do\r''
'/configure: line 29: ` for c in $CCS; do

$ make win
Win64: generating PDF manual
eqn:doc/cntlm.1:0: invalid input character code '13'
eqn:doc/cntlm.1:1: invalid input character code '13'
eqn:doc/cntlm.1:2: invalid input character code '13'
...
```
I am using TortoiseGit 2.7.0.0 and need to find out more about autocrlf...
After cloning the repo those files came out to have DOS line endings; after converting them to Unix line endings, everything works.

**Comment by fralken** [comment]
Thanks for this. Anyway, it appears that in the committed readme files all the line endings have changed. The configuration of autocrlf should be such that the committed file is reverted to the original line endings (that is, unix mode).

**Pull request merged**